packages/common/src/service.js

```diff
@@ -51,14 +51,13 @@ function Service(config, options) {
   });
 
   this.makeAuthenticatedRequest = util.makeAuthenticatedRequestFactory(reqCfg);
-
   this.authClient = this.makeAuthenticatedRequest.authClient;
   this.baseUrl = config.baseUrl;
   this.getCredentials = this.makeAuthenticatedRequest.getCredentials;
   this.globalInterceptors = arrify(options.interceptors_);
   this.interceptors = [];
   this.packageJson = config.packageJson;
-  this.projectId = options.projectId;
+  this.projectId = options.projectId || '{{projectId}}';
```

Comment:
I could be missing parts of the picture still, but why can't we just do something like

```js
this.projectId = options.projectId || this.authClient.projectId;
```

And then handle the error as we did previously?